keylime/tpm2.py

```diff
@@ -390,10 +416,10 @@ def __create_ek(self, asym_alg=None):
                 'opw': owner_pw,
                 'epw': owner_pw
             }
-            if legacy_tools:
+            if tools_version == "3.2":
                 command = "tpm2_getpubek -H 0x81010007 -g {asymalg} -f {ekpubfile} -P {ekpw} -o {opw} -e {epw}".format(**cmdargs)
-            else:
-                command = "tpm2_createek -c - -G {asymalg} -p {ekpubfile} -P {ekpw} -o {opw} -e {epw}".format(**cmdargs)
+            if tools_version == "4.0":
+                command = "tpm2_createek -Q -c 0x81010009 -G {asymalg} -u {ekpubfile} -f pem -p {epw} -w {opw} -P {epw}".format(**cmdargs)
```

Comment:
I would recommend removing `-Q` and leaving this as `-c -` so that a random handle is chosen (instead of hard-coding to `0x81010009`.

Reply:
Also, adding `-f pem` is going to cause problems here.  We get the pem version of the ek in the `__get_pub_ek` method.